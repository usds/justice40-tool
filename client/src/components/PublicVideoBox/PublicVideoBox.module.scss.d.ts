declare namespace PublicVideoBoxNamespace {
    export interface IPublicVideoBoxScss {
        publicVideoContainer: string;
        publicVideoContainerBeta: string;
        publicVideoLink: string;
        youTubeBtn: string;
        buttonContainer: string;
        buttonText: string;
        buttonImage: string;
    }
  }

declare const PublicVideoBoxScssModule: PublicVideoBoxNamespace.IPublicVideoBoxScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: PublicVideoBoxNamespace.IPublicVideoBoxScss;
  };

  export = PublicVideoBoxScssModule;
