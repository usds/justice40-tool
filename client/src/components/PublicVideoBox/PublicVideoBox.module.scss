@use "../../styles/design-system.scss" as *;

.publicVideoContainer {
  @include u-margin-top(3);

  @include at-media-max('desktop'){
    @include u-margin-bottom(3);
  }

  .publicVideoLink {
      display: flex;
      text-decoration: none;

      .youTubeBtn {
        @include u-text("blue-70v");
        @include u-bg("yellow-20v");
        height: 40px;
        @include u-margin-top(3);
        margin-left: auto;
        margin-right: auto;
        width: 14rem;
        
        &:hover {
          @include u-bg("yellow-20");
          @include u-text("gray-90");
        }
        .buttonContainer {
            display: flex;

          .buttonText {
            @include u-margin-right(1);
            }
        
        .buttonImage {
            width: 21px;
            margin-top: -3px;

            filter: invert(13%) sepia(76%) saturate(5142%) hue-rotate(192deg) brightness(80%) contrast(106%);
        }
      }
    }
  }
};

.publicVideoContainerBeta {
  background-color: #F0F0F0;
  border-color: #D8D9DA;
  
  @include u-margin-top(3);

  @include at-media-max('desktop'){
    @include u-margin-bottom(3);
  }

  .publicVideoLink {
      display: flex;
      text-decoration: none;

      .youTubeBtn {
        @include u-text("white");
        @include u-bg("gray-60");
        height: 40px;
        @include u-margin-top(3);
        margin-left: auto;
        margin-right: auto;
        width: 14rem;
        
        &:hover {
          @include u-bg("gray-90");
          @include u-text("gray-10");
        }
        .buttonContainer {
            display: flex;

          .buttonText {
            @include u-margin-right(1);
            }
        
        .buttonImage {
            width: 21px;
            margin-top: -3px;

            filter: invert(100%) sepia(0%) saturate(7489%) hue-rotate(232deg) brightness(101%) contrast(101%);
        }
      }
    }
  }
};