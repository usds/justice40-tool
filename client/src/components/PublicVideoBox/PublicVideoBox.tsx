import React from 'react';
import {Button, SummaryBox, SummaryBoxContent, SummaryBoxHeading} from '@trussworks/react-uswds';
import {useIntl} from 'gatsby-plugin-intl';

import DownloadButton from '../DownloadButton';

import * as styles from './PublicVideoBox.module.scss';
import * as PUBLIC_COPY from '../../data/copy/publicEngage';
import {getDownloadFileUrl} from '../../data/copy/downloads';

// @ts-ignore
import launchIcon from '/node_modules/uswds/dist/img/usa-icons/launch.svg';

interface IPublicVideoBox {
  isBeta: boolean,
  youTubeLink: string,
}

const PublicVideoBox = ({isBeta, youTubeLink}:IPublicVideoBox) => {
  const intl = useIntl();

  return (
    <SummaryBox
      className={isBeta ? styles.publicVideoContainerBeta : styles.publicVideoContainer}>

      <SummaryBoxHeading headingLevel='h2'>
        {isBeta ?
        intl.formatMessage(PUBLIC_COPY.PUBLIC_ENG_VIDEO.TITLE_BETA) :
        intl.formatMessage(PUBLIC_COPY.PUBLIC_ENG_VIDEO.TITLE)}
      </SummaryBoxHeading>

      <SummaryBoxContent>
        {isBeta ? PUBLIC_COPY.RICH_COPY.VIDEO_BOX_BODY2 : PUBLIC_COPY.RICH_COPY.VIDEO_BOX_BODY1}
        <a
          className={styles.publicVideoLink}
          href={youTubeLink}
          target={'_blank'}
          rel="noreferrer"
        >
          <Button
            type="button"
            className={styles.youTubeBtn}
          >
            <div className={styles.buttonContainer}>
              <div className={styles.buttonText}>
                {isBeta ?
                  intl.formatMessage(PUBLIC_COPY.PUBLIC_ENG_VIDEO.BUTTON1_BETA_TEXT) :
                  intl.formatMessage(PUBLIC_COPY.PUBLIC_ENG_VIDEO.BUTTON1_TEXT)
                }
              </div>
              <img
                className={styles.buttonImage}
                src={launchIcon}
                alt={intl.formatMessage(PUBLIC_COPY.PUBLIC_ENG_VIDEO.IMG_ALT_TEXT1)}
              />
            </div>
          </Button>
        </a>
        {isBeta && <DownloadButton
          downloadLink={getDownloadFileUrl(process.env.GATSBY_FILE_DL_PATH_BETA_TRAINING_SLIDES_PPT, true)}
          buttonText={isBeta ?
            intl.formatMessage(PUBLIC_COPY.PUBLIC_ENG_VIDEO.BUTTON2_BETA_TEXT) :
            intl.formatMessage(PUBLIC_COPY.PUBLIC_ENG_VIDEO.BUTTON2_TEXT)
          }
          imageAltTagText={intl.formatMessage(PUBLIC_COPY.PUBLIC_ENG_VIDEO.IMG_ALT_TEXT2)}
          color={isBeta ? 'gray' : 'yellow'}
        />}
      </SummaryBoxContent>
    </SummaryBox>
  );
};

export default PublicVideoBox;
