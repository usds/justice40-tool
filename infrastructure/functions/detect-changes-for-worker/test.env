REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1