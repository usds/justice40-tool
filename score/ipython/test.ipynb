{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
