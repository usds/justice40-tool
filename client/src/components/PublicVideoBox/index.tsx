import PublicVideoBox from './PublicVideoBox';

export default PublicVideoBox;
