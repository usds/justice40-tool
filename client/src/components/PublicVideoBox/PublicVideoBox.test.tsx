import * as React from 'react';
import {render} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import PublicVideoBox from './PublicVideoBox';

describe('rendering of the PublicVideoBox', () => {
  it('checks if component renders when it is in beta', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <PublicVideoBox isBeta={true} />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('checks if component renders when it is NOT in beta', () => {
    const {asFragment} = render(
        <LocalizedComponent>
          <PublicVideoBox isBeta={false} />
        </LocalizedComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
