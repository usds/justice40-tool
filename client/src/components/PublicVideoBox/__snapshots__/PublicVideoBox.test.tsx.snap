// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the PublicVideoBox checks if component renders when it is NOT in beta 1`] = `
<DocumentFragment>
  <div
    class="usa-summary-box"
    data-testid="summary-box"
  >
    <div
      class="usa-summary-box__body"
    >
      <h2
        class="usa-summary-box__heading"
      >
        What's new overview
      </h2>
      <div
        class="usa-summary-box__text"
      >
        
    Watch a recorded update from CEQ on the changes in the 1.0 version of the tool.
  
        <a
          rel="noreferrer"
          target="_blank"
        >
          <button
            class="usa-button"
            data-testid="button"
            type="button"
          >
            <div>
              <div>
                Watch on YouTube
              </div>
              <img
                alt="the icon to show that this button will open in a new tab"
                src="test-file-stub"
              />
            </div>
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering of the PublicVideoBox checks if component renders when it is in beta 1`] = `
<DocumentFragment>
  <div
    class="usa-summary-box"
    data-testid="summary-box"
  >
    <div
      class="usa-summary-box__body"
    >
      <h2
        class="usa-summary-box__heading"
      >
        Need information about the public beta?
      </h2>
      <div
        class="usa-summary-box__text"
      >
        
      Watch a recorded version of the beta version of the CEJST training on YouTube.
  
        <a
          rel="noreferrer"
          target="_blank"
        >
          <button
            class="usa-button"
            data-testid="button"
            type="button"
          >
            <div>
              <div>
                Watch on beta demo
              </div>
              <img
                alt="the icon to show that this button will open in a new tab"
                src="test-file-stub"
              />
            </div>
          </button>
        </a>
        <a
          download=""
          href="//"
        >
          <button
            class="usa-button"
            data-testid="button"
            type="button"
          >
            <div>
              <div>
                Download beta slide deck
              </div>
              <img
                alt="the icon to show that this button will download the file"
                src="test-file-stub"
              />
            </div>
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
